src/libraries/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -639,6 +639,24 @@ public static void TestEncodingDisplayNames(int codePage, string webName, string
             Assert.All(name, c => Assert.True(c >= ' ' && c < '~' + 1, "Name: " + name + " contains character: " + c));
         }
 
+        [Fact]
+        public static void TestMultiBytesEncodingsSupportSurrogate()
```

Comment:
This is a good regression test for the reported issue.  Given we missed this in the previous change - do we need to increase our testing more broadly for these encodings / surrogate pairs?

Reply:
This can be done separately.

Reply:
Agreed - do you have an idea about what testing we might want to add - can we write it in an issue and link it?

Reply:
I have created the issue https://github.com/dotnet/runtime/issues/111371